File: clients/javascript/async/examples/fibonacci.js
#!/usr/bin/env node
// This is free software for the public good of a permacomputer hosted at
// permacomputer.com, an always-on computer by the people, for the people.
// One which is durable, easy to repair, & distributed like tap water
// for machine learning intelligence.
//
// The permacomputer is community-owned infrastructure optimized around
// four values:
//
//   TRUTH      First principles, math & science, open source code freely distributed
//   FREEDOM    Voluntary partnerships, freedom from tyranny & corporate control
//   HARMONY    Minimal waste, self-renewing systems with diverse thriving connections
//   LOVE       Be yourself without hurting others, cooperation through natural law
//
// This software contributes to that vision by enabling code execution across 42+ programming languages through a unified interface, accessible to all.
// Code is seeds to sprout on any abandoned technology.

/**
 * Fibonacci example - standalone version
 *
 * Demonstrates concurrent fibonacci calculations using async/await.
 * Shows how to run multiple concurrent operations with Promise.all().
 *
 * To run:
 *   node fibonacci.js
 *
 * Expected output:
 *   Starting 3 concurrent fibonacci calculations...
 *   [fib-10] Result: fib(10) = 55
 *   [fib-15] Result: fib(15) = 610
 *   [fib-12] Result: fib(12) = 144
 *   All calculations completed!
 */

// Simulated fibonacci calculation (would normally call API)
function fib(n) {
  if (n <= 1) return n;
  return fib(n - 1) + fib(n - 2);
}

async function runFibonacci(n, label) {
  // Simulate async API call delay
  await new Promise((resolve) => setTimeout(resolve, 50));

  const result = fib(n);
  const output = `fib(${n}) = ${result}`;
  console.log(`[${label}] Result: ${output}`);
  return { label, output };
}

async function main() {
  console.log('Starting 3 concurrent fibonacci calculations...');

  // Run all fibonacci calculations concurrently
  const results = await Promise.all([
    runFibonacci(10, 'fib-10'),
    runFibonacci(15, 'fib-15'),
    runFibonacci(12, 'fib-12'),
  ]);

  console.log('All calculations completed!');
  return 0;
}

main().then(process.exit);

File: clients/rust/sync/examples/fibonacci.rs
// PUBLIC DOMAIN - NO LICENSE, NO WARRANTY
//
// Fibonacci example - standalone version
//
// This example demonstrates executing a recursive algorithm.
// Shows how to run complex computations (simulated).
//
// To run:
//     rustc fibonacci.rs && ./fibonacci
//
// Expected output:
//     Executing Fibonacci code...
//     Result status: completed
//     Output:
//     fib(0) = 0
//     fib(1) = 1
//     fib(2) = 1
//     fib(3) = 2
//     fib(4) = 3
//     fib(5) = 5
//     fib(6) = 8
//     fib(7) = 13
//     fib(8) = 21
//     fib(9) = 34
//     fib(10) = 55
//     Execution time: 150ms

fn fib(n: u32) -> u32 {
    if n <= 1 {
        n
    } else {
        fib(n - 1) + fib(n - 2)
    }
}

fn main() {
    // Execute the fibonacci calculation (simulated API call)
    println!("Executing Fibonacci code...");
    println!("Result status: completed");
    println!("Output:");

    for i in 0..=10 {
        println!("fib({}) = {}", i, fib(i));
    }

    println!("Execution time: 150ms");
}
